packages/headless/src/controllers/commerce/core/breadcrumb-manager/headless-core-breadcrumb-manager.test.ts

```diff
@@ -0,0 +1,295 @@
+} from '../../../../features/facets/range-facets/numeric-facet-set/numeric-facet-actions';
+import {buildMockCommerceFacetRequest} from '../../../../test/mock-commerce-facet-request';
+import {
+  buildMockCategoryFacetResponse,
+  buildMockCommerceDateFacetResponse,
+  buildMockCommerceNumericFacetResponse,
+  buildMockCommerceRegularFacetResponse,
+} from '../../../../test/mock-commerce-facet-response';
+import {buildMockCommerceState} from '../../../../test/mock-commerce-state';
+import {
+  buildMockCommerceEngine,
+  MockedCommerceEngine,
+} from '../../../../test/mock-engine-v2';
+import {facetResponseSelector} from '../../product-listing/facets/headless-product-listing-facet-options';
+import {
+  BreadcrumbManager,
+  buildCoreBreadcrumbManager,
+  CoreBreadcrumbManagerOptions,
+  DeselectableValue,
+} from './headless-core-breadcrumb-manager';
+
+jest.mock('../../../../features/facets/facet-set/facet-set-actions');
+jest.mock(
+  '../../../../features/facets/range-facets/numeric-facet-set/numeric-facet-actions'
+);
+jest.mock(
+  '../../../../features/facets/range-facets/date-facet-set/date-facet-actions'
+);
+jest.mock(
+  '../../../../features/facets/category-facet-set/category-facet-set-actions'
+);
+jest.mock(
+  '../../../../features/commerce/product-listing/product-listing-actions'
+);
+jest.mock('../../../../features/breadcrumb/breadcrumb-actions');
+
+describe('core breadcrumb manager', () => {
+  let engine: MockedCommerceEngine;
+  let options: CoreBreadcrumbManagerOptions;
+  let breadcrumbManager: BreadcrumbManager;
+
+  const facetId = 'some_facet_id';
+
+  function initEngine(preloadedState = buildMockCommerceState()) {
+    engine = buildMockCommerceEngine(preloadedState);
+  }
+
+  function initBreadcrumbManager() {
+    breadcrumbManager = buildCoreBreadcrumbManager(engine, options);
+  }
+
+  function setFacetsState({facetId, ...restOfResponse}: AnyFacetResponse) {
+    engine.state.facetOrder.push(facetId);
+    engine.state.commerceFacetSet[facetId] = {
+      request: buildMockCommerceFacetRequest(),
+    };
+    engine.state.productListing.facets.push({facetId, ...restOfResponse});
+  }
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    options = {
+      facetResponseSelector: facetResponseSelector,
+      fetchResultsActionCreator: fetchProductListing,
+    };
+
+    initEngine();
+    initBreadcrumbManager();
+  });
+
+  it('initializes', () => {
+    expect(breadcrumbManager).toBeTruthy();
+  });
+
+  it('adds correct reducers to engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      facetOrder,
+      commerceFacetSet,
+    });
+  });
+
+  it('exposes #subscribe method', () => {
+    expect(breadcrumbManager.subscribe).toBeTruthy();
+  });
+
+  it('#deselectAll deselects all breadcrumbs', () => {
+    breadcrumbManager.deselectAll();
+    expect(deselectAllBreadcrumbs).toHaveBeenCalled();
+  });
+
+  it('#deselectBreadcrumb deselects breadcrumb', () => {
```

Comment:
If you decide not to expose this method after all (see comment here https://github.com/coveo/ui-kit/pull/3746/files#diff-2fea05de14f6b18dbb77d9bbeae8d62a6801951ab09bc225c41a5ca561f9dadcR204), then you can remove this test as well.